File: Code/projects/Languages/C++/calculator.cpp
// Scott Holley
// Calculator in c++

#include <iostream>
// using namespace std is generally not a good idea
// but for this small program, it's okay
using namespace std;


/*
These are functions in c++. They are a little different than in python but the
idea is the same. Make a type of function, and give the parameters types.
In this case, I want to return the operation of the two parameters
*/
float addition(int a, int b){
    return a + b;
}

float subtraction(int a, int b){
    return a - b;
}

float multiplication(int a, int b){
    return a * b;
}

float division(int a, int b){
    return a / b;
}


/*
Every c++ program has to have a main program. In this smaller one, the main
program pretty much does the same as the python main program
 */
int main(){
    int a, b;
    int choice;
    cout << "Enter the first number: ";
    cin >> a;

    cout << "Enter the second number: ";
    cin >> b;

    cout << "Enter the operation\n1)Addition\n2)Subtraction\n3)Multiplication\n4)Division\n";
    cin >> choice;

    if (choice == 1){
        cout << "The answer is: " << addition(a, b) << "\n";
    }
    else if (choice  == 2){
        cout << "The answer is: " << subtraction(a, b) << "\n";       
    }
    else if (choice  == 3){
        cout << "The answer is: " << multiplication(a, b) << "\n";       
    }
    else{
        cout << "The answer is: " << division(a, b) << "\n";
    }
}

File: Code/projects/Languages/GO/calculator.go
// Calculator in Go
package main

import (
	"fmt"
)

func add(a int, b int) int {
	return a + b
}

func sub(a int, b int) int {
	return a - b
}

func mul(a int, b int) int {
	return a * b
}

func main() {
	var num1 int
	var num2 int

	fmt.Print("Please enter the first number: ")
	fmt.Scan(&num1)

	fmt.Print("Please enter the second number: ")
	fmt.Scan(&num2)

	var choice int
	fmt.Print("Please enter the operation:\n1)Addition\n2)Subtraction\n3)Multiplication\n")
	fmt.Scan(&choice)

	if choice == 1 {
		fmt.Print("The answer is: ", add(num1, num2), "\n")
	} else if choice == 2 {
		fmt.Print("The answer is: ", sub(num1, num2), "\n")
	} else {
		fmt.Print("The answer is: ", mul(num1, num2), "\n")
	}

	var cont string
	fmt.Print("Do you want to continue? [Y or N]\n")
	fmt.Scan(&cont)
	check(cont)
}

func check(answer string) {
	if answer == "Y" {
		main()
	}
}
